### PR TITLE
11021/fix/make easier re enable usercomment

### LINKED ISSUE #11021: Make it easier to re-enable user comments
**Is your suggestion for improvement related to a problem? Please describe.**
When viewing the comments for an entry in the Comments tab, there are both "general" comments and user comments, and when user comments are visible, there is a button below prompting "Hide user comments". If that is clicked, the only way to re-enable user comments is through the settings. It is easy to misclick on, and a pain to have to re-enable when that happens.

**Describe the solution you'd like**
It would be nice for a similar button to be available in the same Comments tab to show the user comments back more easily.

**Additional context**

image
image
image

### PR DESCRIPTION
**Pull request by Realmbird**
Closes #11021 (" Make it easier to re-enable user comments #11021")
Makes it easier to re enable user comments by adding show user comments button. User no longer has to go to preferences and show user comments in order to re enable user comments after hiding them.
### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by subhramit** [comment]
Localization tests have failed. Check wherever, during your development, you have used Strings (Sentences). 
Read more at https://devdocs.jabref.org/code-howtos/localization.html.

**Comment by koppor** [comment]
@Realmbird The PR description should be based on the template: https://raw.githubusercontent.com/JabRef/jabref/main/.github/PULL_REQUEST_TEMPLATE.md. It currently misses a reference in the text to the issue. It also misses why this is a draft and what are the next steps. (Not all reviewers read the messages in gitter when reviewing this pull request)

**Comment by koppor** [comment]
We should be more precise here:

## 1. Rename "Show user comments field" to "Show user-specific comment field" (no "s" at the end)

image

Reasons:

1. It is only about the comment field of the **current** user, not any other user
2. It is the field specific for ("tied to") the user - not the general "comment" field

## 2. Rename "Hide user comments" to "Hide user-specific comment field"

image

Reasons: Same as for 1.

## 3. Rename "Show user comments" to "Show user-specific comment field"

image

Reasons: Same as above

## 4. Do NEITHER show "Show user-specific comment field" nor "Hide user-specific comment field" if user-specific comment already shown

Example for user "koppor"

image

image

**Comment by Realmbird** [comment]
@koppor Got all tests to pass and thank you to subhramit  for giving the hint it was a localization problem I just copied Hide\ user\ comments=Hide user comments and changed hide with show. Not sure how this works so where can I why they have \s before works is it to make it into keys?

File: src/main/java/org/jabref/gui/entryeditor/CommentsTab.java [hunk 1]
```diff
@@ -151,6 +151,19 @@ protected void setupPanel(BibEntry entry, boolean compressed) {
             });
             gridPane.add(hideDefaultOwnerCommentButton, 1, gridPane.getRowCount(), 2, 1);
             setCompressedRowLayout();
+        } else {
+            Button showDefaultOwnerCommentButton = new Button(Localization.lang("Show user comments"));
+            showDefaultOwnerCommentButton.setOnAction(e -> {
+                var labelForField = gridPane.getChildren().stream().filter(s -> s instanceof FieldNameLabel).filter(x -> ((FieldNameLabel) x).getText().equals(userSpecificCommentField.getDisplayName())).findFirst();
+                labelForField.ifPresent(label -> gridPane.getChildren().remove(label));
+                fieldEditorForUserDefinedComment.ifPresent(f -> gridPane.getChildren().remove(f.getNode()));
+//                editors.add(userSpecificCommentField);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/entryeditor/CommentsTab.java:160]: No commented-out code

File: src/main/java/org/jabref/gui/entryeditor/CommentsTab.java [hunk 2]
```diff
@@ -151,6 +151,19 @@ protected void setupPanel(BibEntry entry, boolean compressed) {
             });
             gridPane.add(hideDefaultOwnerCommentButton, 1, gridPane.getRowCount(), 2, 1);
             setCompressedRowLayout();
+        } else {
+            Button showDefaultOwnerCommentButton = new Button(Localization.lang("Show user comments"));
```
Comment (koppor) [inline src/main/java/org/jabref/gui/entryeditor/CommentsTab.java:155]: This code is somehow similar to the if branch above. Can you factor out a method with parameters (e.g., the label of the button and whether the usercommentsfield should be shown)

File: src/main/java/org/jabref/gui/entryeditor/CommentsTab.java [hunk 3]
```diff
@@ -151,6 +151,19 @@ protected void setupPanel(BibEntry entry, boolean compressed) {
...
+            showDefaultOwnerCommentButton.setOnAction(e -> {
+                var labelForField = gridPane.getChildren().stream().filter(s -> s instanceof FieldNameLabel).filter(x -> ((FieldNameLabel) x).getText().equals(userSpecificCommentField.getDisplayName())).findFirst();
+                labelForField.ifPresent(label -> gridPane.getChildren().remove(label));
+                fieldEditorForUserDefinedComment.ifPresent(f -> gridPane.getChildren().remove(f.getNode()));
+//                editors.add(userSpecificCommentField);
+
+                entryEditorPreferences.setShowUserCommentsFields(true);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/entryeditor/CommentsTab.java:162]: Please rename the method to `setShowUserSpecificCommentField`

**Review by koppor — CHANGES_REQUESTED** [review]
Please see my comments at https://github.com/JabRef/jabref/pull/11090#issuecomment-2016916456 for a refined requirements list.

**Comment by subhramit** [comment]
> @koppor Got all tests to pass and thank you to @subhramit for giving the hint it was a localization problem I just copied Hide\ user\ comments=Hide user comments and changed hide with show. Not sure how this works so where can I why they have \s before works is it to make it into keys?

In general, in programming backslashes are used to denote escape characters in strings. In the localization file, they are used for special characters and spaces. Using "```\ ```" instead of directly using spaces makes it get recognized as a single key-value pair (key=value form), otherwise spaces will act as separators and make them get interpreted as multiple key-value pairs. Don't exactly remember where I read it, but anybody correct me if I'm wrong.

**Comment by Realmbird** [comment]
Gonna try working on #11090 (comment)
for 4.  Do NEITHER show "Show user-specific comment field" nor "Hide user-specific comment field" if user-specific comment already shown
I am gonna spend some time trying to solve what causes the bug

**Comment by koppor** [comment]
> Gonna try working on #11090 (comment) for 4. Do NEITHER show "Show user-specific comment field" nor "Hide user-specific comment field" if user-specific comment already shown I am gonna spend some time trying to solve what causes the bug

Just store 

    boolean isDefaultOwnerComment = field.equals(userSpecificCommentField);

in a method-local variable `hasDefaultOwnerComment`. Inside the loop, do:

    hasDefaultOwnerComment = hasDefaultOwnerComment || isDefaultOwnerComment ;

Before `// Show "Hide" button`, do 

```java
    // Do not show any buttons if user-specific comment field is present
    if (hasDefaultOwnerComment) {
        return;
    }
```

**Comment by koppor** [comment]
@Realmbird May I ask if you could find time to continue working on this one?

**Comment by Realmbird** [comment]
No I have quit been busy after spring break

**Comment by koppor** [comment]
I think, only little things left. Will try to implement for myself.

**Pull request closed without merging**